domains/runtime/core-evm/src/runtime.rs

```diff
@@ -0,0 +1,797 @@
+use crate::precompiles::FrontierPrecompiles;
+use codec::{Decode, Encode};
+pub use domain_runtime_primitives::{Balance, BlockNumber, Hash, Index, RelayerId};
+use fp_account::EthereumSignature;
+use fp_evm::weight_per_gas;
+use frame_support::dispatch::DispatchClass;
+use frame_support::traits::{ConstU16, ConstU32, Everything, FindAuthor, UnixTime};
+use frame_support::weights::constants::{
+    BlockExecutionWeight, ExtrinsicBaseWeight, ParityDbWeight, WEIGHT_REF_TIME_PER_MILLIS,
+};
+use frame_support::weights::Weight;
+use frame_support::{construct_runtime, parameter_types};
+use frame_system::limits::{BlockLength, BlockWeights};
+use pallet_ethereum::Call::transact;
+use pallet_ethereum::{PostLogContent, Transaction as EthereumTransaction, TransactionStatus};
+use pallet_evm::{
+    Account as EVMAccount, EnsureAccountId20, FeeCalculator, IdentityAddressMapping, Runner,
+};
+use sp_api::impl_runtime_apis;
+use sp_core::crypto::KeyTypeId;
+use sp_core::{Get, OpaqueMetadata, H160, H256, U256};
+use sp_runtime::traits::{
+    AccountIdLookup, BlakeTwo256, Block as BlockT, DispatchInfoOf, Dispatchable, IdentifyAccount,
+    PostDispatchInfoOf, UniqueSaturatedInto, Verify,
+};
+use sp_runtime::transaction_validity::{
+    TransactionSource, TransactionValidity, TransactionValidityError,
+};
+use sp_runtime::{
+    create_runtime_str, generic, impl_opaque_keys, ApplyExtrinsicResult, ConsensusEngineId,
+};
+pub use sp_runtime::{MultiAddress, Perbill, Permill};
+use sp_std::prelude::*;
+use sp_std::time::Duration;
+#[cfg(feature = "std")]
+use sp_version::NativeVersion;
+use sp_version::RuntimeVersion;
+use subspace_runtime_primitives::SHANNON;
+
+/// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
+pub type Signature = EthereumSignature;
+
+/// Some way of identifying an account on the chain. We intentionally make it equivalent
+/// to the public key of our transaction signing scheme.
+pub type AccountId = <<Signature as Verify>::Signer as IdentifyAccount>::AccountId;
+
+/// The address format for describing accounts.
+pub type Address = MultiAddress<AccountId, ()>;
+
+/// Block header type as expected by this runtime.
+pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
+
+/// Block type as expected by this runtime.
+pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+/// A Block signed with a Justification
+pub type SignedBlock = generic::SignedBlock<Block>;
+
+/// BlockId type as expected by this runtime.
+pub type BlockId = generic::BlockId<Block>;
+
+/// The SignedExtension to the basic transaction logic.
+pub type SignedExtra = (
+    frame_system::CheckNonZeroSender<Runtime>,
+    frame_system::CheckSpecVersion<Runtime>,
+    frame_system::CheckTxVersion<Runtime>,
+    frame_system::CheckGenesis<Runtime>,
+    frame_system::CheckMortality<Runtime>,
+    frame_system::CheckNonce<Runtime>,
+    frame_system::CheckWeight<Runtime>,
```

Comment:
Is it intentional to exclude the `ChargeTransactionPayment`? Regular user transactions are still possible due to the pallet balances.

Reply:
yes, I was not sure how this would work with evm. Should be handled in the further PRS